cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -112,10 +113,8 @@ func (ng *nodegroup) DeleteNodes(nodes []*corev1.Node) error {
 	// and we fail fast.
 	replicas := ng.scalableResource.Replicas()
 
-	if replicas-int32(len(nodes)) < 0 {
-		return fmt.Errorf("unable to delete %d machines in %q, machine replicas are < 0 ", len(nodes), ng.Id())
-	} else if replicas-int32(len(nodes)) == 0 && !ng.scalableResource.CanScaleFromZero() {
-		return fmt.Errorf("unable to delete %d machines in %q, machine replicas are 0", len(nodes), ng.Id())
+	if replicas-int32(len(nodes)) <= int32(ng.MinSize()) {
```

Comment:
I think this should be `<` rather than `<=` shouldn't it?

Eg, if I have 3 replicas and want to remove 2, with min size 1, this should be allowed but `3-2 <= 1` is true